pkg/kubevirt/machine.go

```diff
@@ -219,6 +240,80 @@ func (m *Machine) IsReady() bool {
 	return m.hasReadyCondition()
 }
 
+const (
+	defaultCondReason  = "VMNotReady"
+	defaultCondMessage = "VM is not ready"
+)
+
+func (m *Machine) GetVMUnscheduledReason() (reason string, message string) {
+	reason = defaultCondReason
+	message = defaultCondMessage
+
+	if m.vmInstance == nil {
+		return
+	}
+
+	cond := m.getVMCondition(kubevirtv1.VirtualMachineConditionType(corev1.PodScheduled))
+	if cond != nil {
+		if cond.Status == corev1.ConditionTrue {
+			return
+		} else if cond.Status == corev1.ConditionFalse {
+			if cond.Reason == "Unschedulable" {
+				return "Unschedulable", cond.Message
+			}
+		}
+	}
+
+	dvReason, dvMessage, foundDVReason := m.getDVNotProvisionedReason()
+	if foundDVReason {
+		return dvReason, dvMessage
+	}
+
+	return
+}
+
+func (m *Machine) getDVNotProvisionedReason() (string, string, bool) {
+	if m.dataVolume == nil {
+		return "", "", false
+	}
+
+	switch m.dataVolume.Status.Phase {
+	case cdiv1.Succeeded: // DV's OK, return default reason & message
+		return "", "", false
+	case cdiv1.Pending:
+		return "DVPending", fmt.Sprintf("DataVolume %s is pending", m.dataVolume.Name), true
+	case cdiv1.Failed:
+		return "DVFailed", fmt.Sprintf("DataVolume %s failed", m.dataVolume.Name), true
+	default:
+		msg := "DataVolume is not ready"
```

Comment:
maybe add the phase to this default message?

msg: fmt.Sprintf("DataVolume %s is not ready. Phase = %s", m.dataVolume.Name, m.dataVolume.Status.Phase)